api/capi/src/nnstreamer-capi-util.c

```diff
@@ -982,27 +1001,13 @@ _ml_detect_framework (char **model, unsigned int num_models,
   detected =
       gst_tensor_filter_framework_auto_detection ((const gchar **) model,
       num_models);
-  if (detected == NULL) {
-    ml_loge ("The given model has unknown or not supported extension.");
-    return ML_ERROR_INVALID_PARAMETER;
-  }
 
-  if (g_ascii_strcasecmp (detected, "nnfw") == 0) {
-    *nnfw = ML_NNFW_TYPE_NNFW;
-  } else if (g_ascii_strcasecmp (detected, "tensorflow-lite") == 0) {
-    *nnfw = ML_NNFW_TYPE_TENSORFLOW_LITE;
-  } else if (g_ascii_strcasecmp (detected, "vivante") == 0) {
-    *nnfw = ML_NNFW_TYPE_VIVANTE;
-  } else if (g_ascii_strcasecmp (detected, "tensorflow") == 0) {
-    *nnfw = ML_NNFW_TYPE_TENSORFLOW;
-  } else if (g_ascii_strcasecmp (detected, "custom") == 0) {
-    *nnfw = ML_NNFW_TYPE_CUSTOM_FILTER;
-  } else {
-    ml_loge ("The given model has unknown or not supported extension.");
-    status = ML_ERROR_NOT_SUPPORTED;
-  }
+  *nnfw = ml_get_nnfw_type_by_subplugin_name (detected);
```

Comment:
Don't need to check whether `detected` NULL or not?

Reply:
Right! This will be checked in ml_get_nnfw_type_by_subplugin_name().